circuit/src/ecc/same_curve/mod.rs

```diff
@@ -1385,4 +1570,138 @@ mod test {
 		let prover = MockProver::run(k, &test_chip, vec![p_ins]).unwrap();
 		assert_eq!(prover.verify(), Ok(()));
 	}
+
+	#[derive(Clone)]
+	struct EccBatchedMulTestCircuit {
+		points: Vec<UnassignedEcPoint<C, NUM_LIMBS, NUM_BITS, P, EC>>,
+		scalars: Vec<Value<N>>,
+		window_size: u32,
+	}
+
+	impl EccBatchedMulTestCircuit {
+		fn new(
+			points: Vec<EcPoint<C, NUM_LIMBS, NUM_BITS, P, EC>>, scalars: Vec<N>, window_size: u32,
+		) -> Self {
+			Self {
+				points: points.iter().map(|p| UnassignedEcPoint::from(p.clone())).collect(),
+				scalars: scalars.iter().map(|s| Value::known(s.clone())).collect(),
+				window_size,
+			}
+		}
+	}
+
+	impl Circuit<N> for EccBatchedMulTestCircuit {
+		type Config = TestConfig;
+		type FloorPlanner = SimpleFloorPlanner;
+
+		fn without_witnesses(&self) -> Self {
+			Self {
+				points: self
+					.points
+					.iter()
+					.map(|_| UnassignedEcPoint::without_witnesses())
+					.collect(),
+				scalars: self.scalars.iter().map(|_| Value::unknown()).collect(),
+				window_size: u32::default(),
+			}
+		}
+
+		fn configure(meta: &mut ConstraintSystem<N>) -> TestConfig {
+			TestConfig::new(meta)
+		}
+
+		fn synthesize(
+			&self, config: TestConfig, mut layouter: impl Layouter<N>,
+		) -> Result<(), Error> {
+			let mut assigned_scalars_dum = [(); 10].map(|_| None);
+			layouter.assign_region(
+				|| "scalar_mul_values",
+				|region: Region<'_, N>| {
+					let mut ctx = RegionCtx::new(region, 0);
+					for i in 0..self.scalars.len() {
+						let value = ctx.assign_advice(config.common.advice[0], self.scalars[i])?;
+						assigned_scalars_dum[i] = Some(value.clone());
```

Comment:
```suggestion
			let mut assigned_scalars_dum = Vec::new();
			layouter.assign_region(
				|| "scalar_mul_values",
				|region: Region<'_, N>| {
					let mut ctx = RegionCtx::new(region, 0);
					for i in 0..self.scalars.len() {
						let value = ctx.assign_advice(config.common.advice[0], self.scalars[i])?;
						assigned_scalars_dum.push(value);
```

Reply:
The reason I put array in here is when I use `Vec` it doubles the amount of the elements somehow. If our input length is 10 it will put 20 elements in the `Vec` and half of them will be `None`. Do you know what causes this?

Reply:
Ah because the layouter goes through the region twice.
First time to understand the shape: https://github.com/privacy-scaling-explorations/halo2/blob/main/halo2_proofs/src/circuit/floor_planner/single_pass.rs#L93
Second time to assign: https://github.com/privacy-scaling-explorations/halo2/blob/main/halo2_proofs/src/circuit/floor_planner/single_pass.rs#L114

You should declare `assigned_scalars` inside the region itself and then return it at the end

Reply:
We should check the rest of the crate for similar cases